_tools/types/main.go

```diff
@@ -0,0 +1,333 @@
+	// TODO: load package, iterate all structs
+	types := []interface{}{
+		uast.Position{},
+		uast.Positions{},
+		// only embeding GenNode?
+		uast.Identifier{},
+		uast.String{},
+		uast.Bool{},
+		uast.QualifiedIdentifier{},
+		uast.Comment{},
+		uast.Group{},
+		uast.FunctionGroup{},
+		uast.Block{},
+		uast.Alias{},
+		uast.Import{},
+		uast.RuntimeImport{},
+		uast.RuntimeReImport{},
+		uast.InlineImport{},
+		uast.Argument{},
+		uast.FunctionType{},
+		uast.Function{},
+	}
+	for _, typee := range types {
+		out = append(out, uastType{reflect.TypeOf(typee).String()})
+	}
+	fmt.Fprintf(os.Stderr, "%d uast:* types found in SDK\n", len(out))
+	return out
+}
+
+// analyzeFixtures goes though all fixtures, assuming the driver is cloned.
+// It updates given driverStats with results.
+func analyzeFixtures(driver *driverStats) error {
+	fixDir := path.Join(reposRootPath, driver.path, fixtureDir)
+	fixtureFiles, err := lsDir(fixDir)
+	if err != nil && !os.IsNotExist(err) {
+		return err
+	}
+	if os.IsNotExist(err) {
+		driver.skip = true
+		return nil
+	}
+	driver.fixturesNum += len(fixtureFiles)
+
+	re := regexp.MustCompile("(uast:[^\"]*)")
+	for _, file := range fixtureFiles {
+		if strings.HasSuffix(file.Name(), fixtureExt) {
+			fFile, err := os.Open(path.Join(fixDir, file.Name()))
+			if err != nil {
+				log.Fatal(err)
+			}
+			defer fFile.Close()
+
+			scanner := bufio.NewScanner(fFile)
+			for scanner.Scan() {
+				line := scanner.Text()
+				if strings.Contains(line, "uast:") {
+					for _, uastType := range re.FindAllString(line, -1) {
+						driver.fixturesUast[strings.Replace(uastType, ":", ".", 1)]++
+					}
+				}
+			}
+
+			if err := scanner.Err(); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+func lsDir(dir string) ([]os.FileInfo, error) {
+	fmt.Fprintf(os.Stderr, "reading %s/*%s files\n", dir, fixtureExt)
+	f, err := os.Open(dir)
+	if err != nil {
+		return nil, err
+	}
+	files, err := f.Readdir(-1)
+	f.Close()
+	if err != nil {
+		return nil, err
+	}
+	return files, nil
+}
+
+// analyzeCode checks if any of the types are used by
+// this driver's package, though analyzing it's AST.
+func analyzeCode(driver *driverStats, uasts []uastType) {
+	// TODO:
+	// load package
+	// for _, typee := range uasts {
+	//   if typee.isUsedIn(package) {
+	//     driver.codeUast[typee]++
+	//   }
+	// }
+	driver.codeUast["Identifier"]++
+}
+
+func formatMarkdownTable(drivers []*driverStats, uastTypes []uastType) {
+	fmt.Print(header)
+	defer fmt.Print(footer)
+
+	allDrivers := len(drivers)
+	drs := drivers[:0] // filter drivers \wo fixtures
+	for _, x := range drivers {
+		if !x.skip {
+			drs = append(drs, x)
+		}
+	}
+	fmt.Fprintf(os.Stderr, "only %d drivers has fixtures, out of %d\n", len(drs), allDrivers)
+
+	formatMarkdownTableHeader(drs)
+	for _, typee := range uastTypes {
+		// %25s produces nice ASCII result
+		fmt.Printf("|[%s](%s)|", typee.name, goDocUrl+typee.name[strings.IndexRune(typee.name, '.')+1:])
```

Comment:
I think it may be a good idea to also replace `.` with `:` in the table, because it's the type name that we expose to our users in UAST.

Reply:
Not sure I understand, from what I can see, now we expose both to the user:
  - `uast.Identifer` in API [godoc](https://godoc.org/github.com/bblfsh/sdk/uast#Identifier)
  - `uast:Identifier` in text [doc](https://doc.bblf.sh/uast/semantic-uast.html#identifier)

Or do you mean that in the final table, instead of current `uast.*` it's better to use convention of `uast:*`?  

Right now it's kind of consistent in using `.` AND linking to GoDoc, so in case of switching to `:`, would you also suggest to point the link to https://doc.bblf.sh instead?

Reply:
Yes, I think tables should list `uast:Identifier`, but still point to GoDoc (until we have a separate page describing the schema in Babelfish docs). 

The point is that most Babelfish users won't use Go, thus they will only see those types in the context of UAST/XPath (`uast:*`). The links to GoDoc serve as a schema definition and can be in any language. It could be a Protobuf definition or GraphQL for example. We happen to use Go as our DSL.